### PR TITLE
Detect inactive hosts

### PR DESCRIPTION
**Pull request by GiPPe**
https://trello.com/c/8yizIuzx/405-3-detect-inactive-hosts-in-statusboard

### CODE REVIEW

File: appsetup/setup.go [hunk 1]
```diff
@@ -80,3 +82,12 @@ func (a *App) handleEvents() {
 		}
 	}
 }
+
+func (a *App) checkInactiveInstances() {
+	ticker := time.NewTicker(time.Minute * time.Duration(a.Cnf.Storage.InactivityCheck))
+	go func() {
+		for _ = range ticker.C {
```
Comment (kostyantyn) [inline appsetup/setup.go]: can be written `for range ticker.C {`

File: config.example.yml [hunk 2]
```diff
@@ -6,7 +6,10 @@ dashboard:
 # Easy to setup because there are no external dependencies
 # however can be used only on 1 machine otherwise data will be
 # unsynchronized. Also all data are lost when process is restarted.
-storage: memory
+storage:
+  type: 'memory'
+  inactivity_period_in_min: 45
+  inactivity_check_in_min: 15
```
Comment (kostyantyn) [inline config.example.yml]: i think it's enough to have 1 parameter. second one can be hardcoded in a code. I suggest to use `1 min`

File: sources/httppush/handler.go [hunk 3]
```diff
@@ -34,17 +35,17 @@ func NewHTTPPush(channel chan sources.Event, logger *log.Logger, addr string) *H
 			return
 		}
 
-		status := params.Get("status")
-		if status != "1" && status != "0" {
+		status, err := strconv.Atoi(params.Get("status"))
```
Comment (kostyantyn) [inline sources/httppush/handler.go]: let's validate that status is in a valid range `0`, `1` or `2`

File: storages/memorystorage/board.go [hunk 4]
```diff
@@ -45,9 +56,27 @@ func (b *Board) RemoveInstance(instanceIP string) bool {
...
 }
 
+func (b *Board) RemoveInactiveInstances(inactivityPeriod int) {
+	b.mx.Lock()
+	defer b.mx.Unlock()
+
+	for instanceIP, instance := range b.instances {
+		lastUpdateInterval := time.Unix(0, time.Now().UnixNano()).Sub(time.Unix(0, instance.LastEventReceivedAt))
+		if int(math.Floor(lastUpdateInterval.Minutes())) > inactivityPeriod {
+			delete(b.instances, instanceIP)
```
Comment (kostyantyn) [inline storages/memorystorage/board.go]: we should also remove instance from services

**Comment by kostyantyn** [comment]
let's define constants in storage for every status type
```
const(
  DownStatus = iota
  UpStatus
  InactiveStatus
)
```

File: sources/httppush/handler.go (hunk 3 above)
Comment (GiPPe) [inline sources/httppush/handler.go]: Why 2? nobody sends 2 so far

File: storages/memorystorage/board.go (hunk 4 above)
Comment (GiPPe) [inline storages/memorystorage/board.go]: that's right. I am gonna fix this and make the necessary refactoring

**Comment by GiPPe** [comment]
@kostyantyn If I remove the instance from the service, it does not make sense to set status as inactive once the instance is removed.

File: config.example.yml (hunk 2 above)
Comment (GiPPe) [inline config.example.yml]: Isn't 1 minute too short? Should we divide inactivity_period_in_min by 3 maybe? @kostyantyn WDYT?

Comment (GiPPe) [inline config.example.yml]: @kostyantyn we are gonna set it to 1 minute then.

**Pull request merged**